starfish/imagestack/imagestack.py

```diff
@@ -699,6 +696,12 @@ def transform(
         n_processes : Optional[int]
             The number of processes to use for apply. If None, uses the output of os.cpu_count()
             (default = None).
+        finalizer_callable : Callable
```

Comment:
Missing: why someone would want to use this, and examples of usage.

Reply:
`func` gets the sliced data.  `finalizer_callable` gets the original data array, the results, and the slice list.